app/action/responsibility/post.controller.js

```diff
@@ -0,0 +1,35 @@
+const { body } = require('express-validator')
+
+const errorMessage = 'Select the person responsible'
+const options = ['SERVICE_USER', 'PRACTITIONER', 'OTHER']
+
+const validationRules = () => {
+  return [
+    body('responsibility', errorMessage)
+      .not()
+      .isEmpty()
+      .bail()
+      .if(responsibility => typeof responsibility === 'string')
+      .isString()
+      .isIn(options),
+    body('responsibility', errorMessage)
+      .if(responsibility => Array.isArray(responsibility))
```

Comment:
Dimness again. Why check if it's an array if the next test is the built in check to see if it's an array?

Reply:
The 'If' statement does not throw the error, it provides an opportunity to bail out of the current validation stream without reporting an error scenario.In this scenario, the type of Array is implied by continuing through the `if` however the further validation provided by the `isArray` statement is still required. The `isArray` in this instance checks for an array with a length of between 1 and 3, although I don't think it would/could ever receive an Array of length 1.